source/common/network/socket_option_impl.cc

```diff
@@ -32,8 +32,25 @@ bool SocketOptionImpl::setOption(Socket& socket, Socket::SocketState state) cons
         return false;
       }
     }
-  }
+  } else if (state == Socket::SocketState::Listening) {
```

Comment:
Please wrap the `IP_TRANSPARENT` option setting above into `if (state != Socket::SocketState::Listening) { ... }`.

Reply:
Good catch!